### PR TITLE
Add Argument overload with CLR type

### PR DESCRIPTION
**Pull request by Shane32**
Complements:
- #3247

### CODE REVIEW

File: src/GraphQL.ApiTests/net6/GraphQL.approved.txt [hunk 1]
```diff
@@ -1152,7 +1156,6 @@ namespace GraphQL.Execution
     public class ExecutionStrategyRegistration : System.IEquatable<GraphQL.Execution.ExecutionStrategyRegistration>
     {
         public ExecutionStrategyRegistration(GraphQL.Execution.IExecutionStrategy Strategy, GraphQLParser.AST.OperationType Operation) { }
-        protected virtual System.Type EqualityContract { get; }
         public GraphQLParser.AST.OperationType Operation { get; set; }
```
Comment (Shane32) [inline src/GraphQL.ApiTests/net6/GraphQL.approved.txt:1159]:
```suggestion
        protected virtual System.Type EqualityContract { get; }
        public GraphQLParser.AST.OperationType Operation { get; set; }
```

Comment (Shane32) [inline src/GraphQL.ApiTests/net6/GraphQL.approved.txt:1159]:
```suggestion
        protected virtual System.Type EqualityContract { get; }
        public GraphQLParser.AST.OperationType Operation { get; set; }
```

File: src/GraphQL.ApiTests/netstandard20+netstandard21/GraphQL.approved.txt [hunk 2]
```diff
@@ -1152,7 +1156,6 @@ namespace GraphQL.Execution
     public class ExecutionStrategyRegistration : System.IEquatable<GraphQL.Execution.ExecutionStrategyRegistration>
     {
         public ExecutionStrategyRegistration(GraphQL.Execution.IExecutionStrategy Strategy, GraphQLParser.AST.OperationType Operation) { }
-        protected virtual System.Type EqualityContract { get; }
         public GraphQLParser.AST.OperationType Operation { get; set; }
```
Comment (Shane32) [inline src/GraphQL.ApiTests/netstandard20+netstandard21/GraphQL.approved.txt:1159]:
```suggestion
        protected virtual System.Type EqualityContract { get; }
        public GraphQLParser.AST.OperationType Operation { get; set; }
```

File: src/GraphQL/Builders/FieldBuilder.cs [hunk 3]
```diff
@@ -203,6 +203,14 @@ protected FieldBuilder(FieldType fieldType)
                 configure?.Invoke(arg);
             });
 
+        /// <summary>
+        /// Adds an argument to the field.
+        /// </summary>
+        /// <typeparam name="TArgumentGraphType">The graph type of the argument.</typeparam>
+        /// <param name="name">The name of the argument.</param>
+        public virtual FieldBuilder<TSourceType, TReturnType> Argument<TArgumentGraphType>(string name)
+            where TArgumentGraphType : IGraphType => Argument(typeof(TArgumentGraphType), name, null);
```
Comment (Shane32) [inline src/GraphQL/Builders/FieldBuilder.cs:212]:
```suggestion
            where TArgumentGraphType : IGraphType
            => Argument(typeof(TArgumentGraphType), name, null);
```

File: src/GraphQL/Builders/FieldBuilder.cs [hunk 4]
```diff
@@ -212,6 +221,44 @@ protected FieldBuilder(FieldType fieldType)
...
+
+        /// <summary>
+        /// Adds an argument to the field.
+        /// </summary>
+        /// <typeparam name="TArgumentClrType">The clr type of the argument.</typeparam>
+        /// <param name="name">The name of the argument.</param>
+        /// <param name="nullable">Indicates if the argument is optional or not.</param>
+        /// <param name="description">The description of the argument.</param>
+        /// <param name="configure">A delegate to further configure the argument.</param>
+        public virtual FieldBuilder<TSourceType, TReturnType> Argument<TArgumentClrType>(string name, bool nullable, string? description, Action<QueryArgument>? configure = null)
```
Comment (sungam3r) [inline src/GraphQL/Builders/FieldBuilder.cs:255]: See above:
```
[Obsolete("Please use Action<QueryArgument> parameter from other Argument() method overloads to set default value for parameter or use Arguments() method. This method will be removed in v8.")]
        public virtual FieldBuilder<TSourceType, TReturnType> Argument<TArgumentGraphType, TArgumentType>(string name, string? description,
            TArgumentType? defaultValue = default, Action<QueryArgument>? configure = null)
```

I would like to minimize API surface here too as well as I did for `FieldXXX` methods.

**Review by sungam3r — APPROVED** [review]

File: src/GraphQL/Builders/FieldBuilder.cs (hunk 4 above)
Comment (Shane32) [reply src/GraphQL/Builders/FieldBuilder.cs:255]: I matched what you had done previously -- that is why there is no overload with a `defaultValue` param. Lmk if there's anything else you'd like done.

Comment (sungam3r) [reply src/GraphQL/Builders/FieldBuilder.cs:255]: Ah, I see existing overload with description. I agree that `description` property is more widely used (I'm sure) than `defaultValue`. I'm fine to keep it as is for now.

**Pull request merged**